pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -552,8 +567,7 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 		if err != nil {
 			return summary, err
 		}
-		valueScratch = append(valueScratch[:0], v...)
-		value, err := storage.DecodeValueFromMVCCValue(valueScratch)
```

Comment:
hmm this change means that we need to unmarshall the mvcc value header during all restores in this hot path. Instead, under the `if rd.setLDRSource` branch, could we contain all the logic for constructing a new `mvccValue` buf that we pass to the `batcher.AddMVCCKey` call?  If theres a bit of repeat code around checking and initing the checksum, that seems fine.

Reply:
alternatively, you could create a new `AddMVCCWithLDRData` method to the sst_batcher, which conducts this extra work and wraps `AddMVCCKey`